bindings/java/pom.xml

```diff
@@ -0,0 +1,117 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.ucx.jucx</groupId>
+  <artifactId>jucx</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <packaging>jar</packaging>
+
+  <name>jucx</name>
+  <url>https://github.com/openucx/ucx</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <native.dir>${basedir}/src/main/native</native.dir>
+    <ucx.src.dir>${basedir}/../../src</ucx.src.dir>
+    <ucm.lib.path>${ucx.src.dir}/ucm/.libs</ucm.lib.path>
+    <ucs.lib.path>${ucx.src.dir}/ucs/.libs</ucs.lib.path>
+    <uct.lib.path>${ucx.src.dir}/uct/.libs</uct.lib.path>
+    <ucp.lib.path>${ucx.src.dir}/ucp/.libs</ucp.lib.path>
+  </properties>
+
+  <dependencies>
+
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.7.25</version>
```

Comment:
Specific version.
Why do we need the log4j API or logging in general?

Reply:
let's avoid external runtime dependencies, especially of specific versions. This code is supposed to work on multiple different platforms.

Reply:
Maven will be pulling all these dependencies into the .m2 database at compile time. Then, if someone is pulling this into a Java project for deployment, they should be setting up their pom files to pack up all the dependency jars into their final product automagically based on the pom files in our project. Final product then runs anywhere with a compatible JRE and the UCX libraries where it can find them.

Probably avoiding specific versions is not a bad idea, since this makes it more likely that it can use the same log4j versions as other stuff in the final project -> smaller final bundle of jars to distribute. Also, if someone is building this somewhere without internet access, they will need a maven mirror or package cache setup, so avoiding specific version will make it more likely their local maven will have what they need.

Reply:
@gmegan I was thinking about including slf4j with an [optional](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html) tag, then the user will be able to define which implementation and version of slf4j he wants to use. How does that sound? should I do the same for junit?

I'm still not sure what to do about the plugins I use, should I use a version range, i.e.
```xml
    <artifactId>maven-antrun-plugin</artifactId>
    <version>[1.2,1.8]</version>
```
Any other ideas?

Reply:
Digging around some more about the slf4j version issues shows lots of warnings from the package maintainers to make sure all versions of slf4j-binding match within a package. So, I think optional flag is a bit more risky than just putting a version, and setting up the pom to let someone override the version in their own maven settings.xml.

For more flexible package resolution, I would think setting a min version is fine. The slf4j maintainers notes say they are pretty serious about backward compatibility.

So here is min version with a property that can be overridden by a user defined settings.xml property.

```
<properties>
        <slf4jVersion>(1.7.25,)</slf4jVersion>
</properties>

<dependencies>
    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-log4j12</artifactId>
      <version>${slf4jVersion}</version>
    </dependency>
```

Reply:
I wouldn't worry too much about the maven-* plugin versions as long as they are reasonably recent. Maven is going to pull in a ton of transient build deps in any case. When I build this package locally, I am getting 84 jar files in my .m2, just from this one pom.xml. Majority are just for the build, and don't need to end up in a final product. So I would just put down a fixed version that I know works.

Reply:
Since junit is just in the testing scope, I would suggest leave it as a fixed version. It isn't going to get pulled into other projects as a transitive dependency anyway. It could optionally be added as a property to allow user override.